src/util/BitField.js

```diff
@@ -144,7 +145,10 @@ class BitField {
     if (typeof defaultBit === typeof bit && bit >= defaultBit) return bit;
     if (bit instanceof BitField) return bit.bitfield;
     if (Array.isArray(bit)) return bit.map(p => this.resolve(p)).reduce((prev, p) => prev | p, defaultBit);
-    if (typeof bit === 'string' && typeof this.FLAGS[bit] !== 'undefined') return this.FLAGS[bit];
+    if (typeof bit === 'string') {
+      if (typeof this.FLAGS[bit] !== 'undefined') return this.FLAGS[bit];
+      if (!isNaN(bit)) return BigInt(bit);
```

Comment:
This isn't accurate as not every bitfield is a bigint. It should be done on the class that needs it, no?

Reply:
In the typedoc it is explicitly said that the value can be a BigInt, but your right, this is not the case for all bitFields.
There would therefore be two solutions:
  - Cast the different types according to an `instanceof` of` defaultBit`.
  - Extends all static `resolve` functions in all classes that use a different type than the default in the `BitField` class.
 
I would prefer the first one.
The utility of the `defaultBit` value is also to tell to ` BitField` from its extended class what the type of the bitfield is.
I would find it off-putting to have to modify the base function of each class when it can be done once in the `resolve`.

Reply:
Something like this I guess:
```js
if (!isNaN(bit)) return (typeof defaultBit === 'bigint' ? BigInt : Number)(bit);
// or
if (!isNaN(bit)) return typeof defaultBit === 'bigint' ? BigInt(bit) : Number(bit);
```

Reply:
I checked all BitField extended classes of discord.js and it seems that only Permissions use bigint.
So it should not be painful to extend `resolve` once.
The two options would be good I guess.

Reply:
Since we already check the type of the default bit in `toJSON` it's probably fine to go with the first option.